src/txn/txn_timestamp.c

```diff
@@ -198,24 +198,29 @@ __wt_txn_parse_timestamp(WT_SESSION_IMPL *session, const char *name,
  */
 static int
 __txn_get_pinned_timestamp(
-   WT_SESSION_IMPL *session, wt_timestamp_t *tsp, bool include_checkpoint)
+   WT_SESSION_IMPL *session, wt_timestamp_t *tsp, bool include_checkpoint,
+   bool include_oldest)
 {
 	WT_CONNECTION_IMPL *conn;
 	WT_TXN *txn;
 	WT_TXN_GLOBAL *txn_global;
 
 	conn = S2C(session);
 	txn_global = &conn->txn_global;
+	__wt_timestamp_set_zero(tsp);
```

Comment:
This has the side effect that if the caller gets back `WT_NOTFOUND` that the caller's timestamp has been zeroed. That is a new side effect that it didn't do before. Perhaps move it as an `else` clause at about line 216 below instead.

Reply:
Sure, done.